quicklogic/common/toolchain_wrappers/ql_symbiflow

```diff
@@ -499,17 +499,22 @@ else
     PCF_MAKE="\${current_dir}/build/${TOP}_dummy.pcf"
 fi
 
+PROCESS_SDC=`realpath ${MYPATH}/python/process_sdc_constraints.py`
 if ! [ -z "$SDC" ]; then
-	if ! [ -f "$SOURCE"/$SDC ];then
-            echo "The sdc file: $SDC is missing at: $SOURCE"
-            exit 1
-         else
-	    SDC_MAKE="\${current_dir}/$SDC"
-	    cp ${CURR_DIR}/$SDC ${CURR_DIR}/build/${TOP}.sdc
-         fi
+    if ! [ -f "$SOURCE"/$SDC ];then
+        echo "The sdc file: $SDC is missing at: $SOURCE"
+        exit 1
+    else
+        SDC_MAKE="\${current_dir}/build/${TOP}.sdc"
+        if [ -f "$SOURCE"/$PCF_FILE ];then
+            python3 ${PROCESS_SDC} --sdc-in ${CURR_DIR}/${SDC} --pcf ${CURR_DIR}/${PCF} --sdc-out ${CURR_DIR}/build/${TOP}.sdc
```

Comment:
This command is called only if `ql_symbiflow` is invoked. I wonder if this might belong in one of the toolchain, but I think it is ok to leave it here for the time being.

Reply:
Right, I think that for now having device pins in SDC is required only by QuickLogic QLF architecture(s)